src/main.cpp

```diff
@@ -351,6 +369,19 @@ bool skip_die(const dies& dies, die& d, const std::string_view& symbol) {
 
 /**************************************************************************************************/
 
+void update_progress() {
+    if (!settings::instance()._show_progress) return;
+
+    std::size_t done = globals::instance()._die_analyzed_count;
+    std::size_t total = globals::instance()._die_processed_count;
+    std::size_t percentage = static_cast<double>(done) / total * 100;
+
+    std::cout << '\r' << done << "/" << total << "  " << percentage << "%; ";
+    std::cout << globals::instance()._odrv_count << " violation(s) found";
```

Comment:
I think you'll need to overprint trailing space here. Where the percentage goes down it leaves semicolons on the command line.

Reply:
Added, thanks.